src/gridworld.jl

```diff
@@ -0,0 +1,99 @@
+const GWPos = SVector{2,Int}
+
+@with_kw struct SimpleGridWorld <: MDP{GWPos, Symbol}
+    size::Tuple{Int, Int}           = (10,10)
+    rewards::Dict{GWPos, Float64}   = Dict(GWPos(4,3)=>-10.0, GWPos(4,6)=>-5.0, GWPos(9,3)=>10.0, GWPos(8,8)=>3.0)
+    terminate_in::Set{GWPos}        = Set((GWPos(4,3), GWPos(4,6), GWPos(9,3), GWPos(8,8)))
+    tprob::Float64                  = 0.7
+    discount::Float64               = 0.95
+end
+
+
+# States
+
+function POMDPs.states(mdp::SimpleGridWorld)
+    ss = vec(GWPos[GWPos(x, y) for x in 1:mdp.size[1], y in 1:mdp.size[2]])
+    push!(ss, GWPos(-1,-1))
+    return ss
+end
+POMDPs.n_states(mdp::SimpleGridWorld) = prod(mdp.size) + 1
+function POMDPs.stateindex(mdp::SimpleGridWorld, s::AbstractVector{Int})
+    if all(s.>0)
+        return LinearIndices(mdp.size)[s...]
+    else
+        return n_states(mdp)
+    end
+end
+
+POMDPs.initialstate(mdp::SimpleGridWorld, rng::AbstractRNG) = GWPos(rand(rng, 1:mdp.size[1]), rand(rng, 1:mdp.size[2]))
+
+
+# Actions
+
+POMDPs.actions(mdp::SimpleGridWorld) = (:up, :down, :left, :right)
+Base.rand(rng::AbstractRNG, t::NTuple{L,Symbol}) where L = t[rand(rng, 1:length(t))] # don't know why this doesn't work out of the box
+POMDPs.n_actions(mdp::SimpleGridWorld) = 4
+
+const dir = Dict(:up=>GWPos(0,1), :down=>GWPos(0,-1), :left=>GWPos(-1,0), :right=>GWPos(1,0))
+const aind = Dict(:up=>1, :down=>2, :left=>3, :right=>4)
+
+POMDPs.actionindex(mdp::SimpleGridWorld, a::Symbol) = aind[a]
+
+
+
+# Transitions
+
+POMDPs.isterminal(m::SimpleGridWorld, s::AbstractVector{Int}) = any(s.<0)
+
+
+function POMDPs.transition(mdp::SimpleGridWorld, s::AbstractVector{Int}, a::Symbol)
+    if isterminal(mdp, s) || s in mdp.terminate_in
```

Comment:
Hmm. Shouldn't `isterminal` handle both these cases?

Reply:
terminate_in is the set of states that transition to a terminal state *on the next timestep*

Reply:
could make that clearer, maybe it should be called `terminate_from`

Reply:
or switching the order ... or maybe I should just add docs :P